ksqldb-engine/src/main/java/io/confluent/ksql/query/QueryRegistryImpl.java

```diff
@@ -125,9 +126,7 @@ private QueryRegistryImpl(final QueryRegistryImpl original) {
         .filter(Optional::isPresent)
         .map(Optional::get)
         .collect(Collectors.toList());
-    streams.addAll(original.streams.stream()
-        .map(SandboxedSharedKafkaStreamsRuntimeImpl::new)
-        .collect(Collectors.toList()));
+    sourceStreams.addAll(original.streams);
```

Comment:
We used to copy these over when the sandbox was made. But the sandbox isn't always used for persistant queries and isn't cleaned up so we should wait to copy them over until we are sure they are needed and will be cleaned up

Reply:
I'm wondering: since we now always `closeRuntimes` at the end even if no persistent queries are created during the lifetime of the sandbox, does that still matter to do the late-copy and keep a separate `sourceStreams` from `streams`?

Reply:
We actually don't always close the sandbox. the only persistant resource used in the sandbox that I can tell is the runtimes. So if we don't late copy the sourceStreams they will be left behind quite often